Makefile

```diff
@@ -1,21 +1,45 @@
-.PHONY: build-rust-optimizer build-workspace-optimizer build publish-rust-optimizer publish-workspace-optimizer publish
+.PHONY: build-rust-optimizer build-workspace-optimizer build create-rust-optimizer-multi use-rust-optimizer-multi publish-rust-optimizer publish-workspace-optimizer publish
 
 DOCKER_NAME_RUST_OPTIMIZER := "cosmwasm/rust-optimizer"
 DOCKER_NAME_WORKSPACE_OPTIMIZER := "cosmwasm/workspace-optimizer"
-DOCKER_TAG := 0.12.1
+DOCKER_TAG := 0.12.2
 
-build-rust-optimizer:
-	docker build -t $(DOCKER_NAME_RUST_OPTIMIZER):$(DOCKER_TAG) --target rust-optimizer .
+# Native arch
+BUILDARCH := $(shell uname -m)
 
-build-workspace-optimizer:
-	docker build -t $(DOCKER_NAME_WORKSPACE_OPTIMIZER):$(DOCKER_TAG) --target workspace-optimizer .
+# Build multi-CPU architecture images and publish them. rust alpine images support the linux/amd64 and linux/arm64/v8 architectures.
+build: build-rust-optimizer build-workspace-optimizer
 
-publish-rust-optimizer: build-rust-optimizer
-	docker push $(DOCKER_NAME_RUST_OPTIMIZER):$(DOCKER_TAG)
+create-rust-optimizer-multi:
+	docker context create rust-optimizer-multi
+	docker buildx create --name rust-optimizer-multi rust-optimizer-multi
 
-publish-workspace-optimizer: build-workspace-optimizer
-	docker push $(DOCKER_NAME_WORKSPACE_OPTIMIZER):$(DOCKER_TAG)
+use-rust-optimizer-multi:
+	$(MAKE) create-rust-optimizer-multi || true
+	docker buildx use rust-optimizer-multi
 
-build: build-rust-optimizer build-workspace-optimizer
+build-rust-optimizer-x86_64: use-rust-optimizer-multi
+	docker buildx build --platform linux/amd64 -t $(DOCKER_NAME_RUST_OPTIMIZER):$(DOCKER_TAG) --target rust-optimizer --load .
+
+build-rust-optimizer-arm64: use-rust-optimizer-multi
+	docker buildx build --platform linux/arm64/v8 -t $(DOCKER_NAME_RUST_OPTIMIZER):$(DOCKER_TAG) --target rust-optimizer --load .
+
+build-workspace-optimizer-x86_64: use-rust-optimizer-multi
+	docker buildx build --platform linux/amd64 -t $(DOCKER_NAME_WORKSPACE_OPTIMIZER):$(DOCKER_TAG) --target workspace-optimizer --load .
+
+build-workspace-optimizer-arm64: use-rust-optimizer-multi
+	docker buildx build --platform linux/arm64 -t $(DOCKER_NAME_WORKSPACE_OPTIMIZER):$(DOCKER_TAG) --target workspace-optimizer --load .
```

Comment:
Any reason why this is `--platform linux/arm64` instead of `--platform linux/arm64/v8`? Which of the two is the one we want?

Reply:
They are currently synonyms. Better to use `arm64/v8` IMO, which is more specific.